src/connection.js

```diff
@@ -1696,6 +1808,47 @@ Connection.prototype.STATE = {
       }
     }
   },
+  SENT_LOGIN7_WITH_FEDAUTH: {
+    name: 'SentLogin7Withfedauth',
+    events: {
+      socketError: function() {
+        return this.transitionTo(this.STATE.FINAL);
+      },
+      connectTimeout: function() {
+        return this.transitionTo(this.STATE.FINAL);
+      },
+      data: function(data) {
+        if (this.fedAuthInfo.responsePending) {
+          // We got data from the server while we're waiting for adal authentication context
+          // call to complete on the client. We cannot process server data
+          // until this call completes as the state can change on completion of
+          // the call. Queue it for later.
+          const boundDispatchEvent = this.dispatchEvent.bind(this);
+          return setImmediate(boundDispatchEvent, 'data', data);
+        } else {
+          return this.sendDataToTokenStreamParser(data);
```

Comment:
I think we don't need this `if` condition in `data` event. As adal library will be called after `fedAuthInfo` event gets triggered (which happens after `SENT_LOGIN7_WITH_FEDAUTH.data` event), it makes sense to have this condition just in `message` event.